### DISCUSSION TITLE
Impact of pinning after updating to Testing

### DISCUSSION THREAD

**Discussion by bin-linux** [SpiralLinux/SpiralLinux-project#106] [Q&A]
Just a bit of forward planning here. I've updated one of my Spiral VMs to MATE testing - naturally all went well.
However I am wondering out the impact of the kernel pinning. For example, when Bookworm goes live it will then be possible to install newer kernels from future backports, but won't that be messed up by inherited pinning? Also with 5.18 going end of  life there will surely have to be a shift somewhen?
EDIT: After upgrading I notice that the kernel pinning is gone. This begs the question of what will happen in future. If the aim of Spiral is to stay on backports and fast-track then when Bookworm goes live the repos will have to be updated - I assume manually. How will the kernel pinning be re-instated?

**Comment by geckolinux** [comment]
Hi there, sorry for the slow reply.
When upgrading according to the SpiralLinux wiki instructions there are two possibilities:

1. [Upgrade to Testing](https://github.com/SpiralLinux/SpiralLinux-project/wiki#switching-from-debian-stable-to-the-testing-or-unstable-branch) (currently Bookworm) or Unstable. In this case the Backports repository should be disabled, so although the `99backports` pinning file is still in place in won't have any effect because `bullseye-backports` is no longer available:

```
Package: linux-image-*
Pin: release a=bullseye-backports
Pin-Priority: 600
```

2. [Upgrading to the next Debian Stable release](https://github.com/SpiralLinux/SpiralLinux-project/wiki#upgrading-to-a-major-new-debian-stable-release) (after its official release). In this case I should include in the documentation that the name of the `release a=` line should be changed to the next Debian Stable name if the Backports repo is still desired. Thanks for calling this to my attention.
